lib/codecs/TypedLiteralEncoder.js

```diff
@@ -0,0 +1,35 @@
+/*!
+ * Copyright (c) 2024 Digital Bazaar, Inc. All rights reserved.
+ */
+import {Token, Type} from 'cborg';
+import {CborldEncoder} from './CborldEncoder.js';
+
+export class TypedLiteralEncoder extends CborldEncoder {
+  constructor({value, compressionMap} = {}) {
+    super();
+    this.value = value;
+    this.compressionMap = compressionMap;
+  }
+
+  encode() {
+    const {value} = this;
+    const id = this.compressionMap.get(value);
+    if(id === undefined) {
+      return new Token(Type.string, value);
```

Comment:
Hmm, is this right? What if the value is not a string? Should it just not use this encoder instead? We might want a different `createEncoder` implementation below that does this map check and just returns undefined for the encoder (implicitly) if there is no compression map or the compression map does not have the value.